File: scripts/read-excel.js
#!/usr/bin/env node

const XLSX = require('xlsx');
const path = require('path');

const EXCEL_FILE = 'e:\\SEP490\\Performance_Test_Report.xlsx';

try {
  console.log('📖 Reading Excel file...\n');
  
  const workbook = XLSX.readFile(EXCEL_FILE);
  
  console.log('📊 Workbook Sheets:');
  console.log(workbook.SheetNames);
  console.log(`\nTotal Sheets: ${workbook.SheetNames.length}\n`);
  
  // Read each sheet
  workbook.SheetNames.forEach((sheetName, index) => {
    console.log(`\n${'='.repeat(80)}`);
    console.log(`Sheet ${index + 1}: ${sheetName}`);
    console.log('='.repeat(80));
    
    const worksheet = workbook.Sheets[sheetName];
    const data = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
    
    // Display first 20 rows
    const rowsToShow = Math.min(20, data.length);
    console.log(`Showing ${rowsToShow} of ${data.length} rows:\n`);
    
    data.slice(0, rowsToShow).forEach((row, i) => {
      console.log(`Row ${i + 1}: ${JSON.stringify(row)}`);
    });
    
    if (data.length > 20) {
      console.log(`\n... (${data.length - 20} more rows)`);
    }
  });
  
  console.log('\n' + '='.repeat(80));
  console.log('✅ File read successfully');
  
} catch (error) {
  console.error('❌ Error reading Excel file:');
  console.error(error.message);
  process.exit(1);
}

File: scripts/read-excel.py
import openpyxl
import sys

excel_file = r'e:\SEP490\Performance_Test_Report.xlsx'

try:
    print('📖 Reading Excel file...\n')
    
    workbook = openpyxl.load_workbook(excel_file)
    
    print('📊 Workbook Sheets:')
    print(workbook.sheetnames)
    print(f'\nTotal Sheets: {len(workbook.sheetnames)}\n')
    
    # Read each sheet
    for sheet_idx, sheet_name in enumerate(workbook.sheetnames):
        print('\n' + '='*80)
        print(f'Sheet {sheet_idx + 1}: {sheet_name}')
        print('='*80)
        
        worksheet = workbook[sheet_name]
        
        # Get data
        rows_list = []
        for row in worksheet.iter_rows(values_only=True):
            rows_list.append(row)
        
        # Display first 25 rows
        rows_to_show = min(25, len(rows_list))
        print(f'Showing {rows_to_show} of {len(rows_list)} rows:\n')
        
        for i, row in enumerate(rows_list[:rows_to_show]):
            print(f'Row {i + 1}: {row}')
        
        if len(rows_list) > 25:
            print(f'\n... ({len(rows_list) - 25} more rows)')
    
    print('\n' + '='*80)
    print('✅ File read successfully')
    
except Exception as error:
    print(f'❌ Error reading Excel file:')
    print(f'{error}')
    sys.exit(1)
